plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2936,9 +2940,20 @@ public MaterializedViewFreshness getMaterializedViewFreshness(ConnectorSession s
         String dependsOnTables = Optional.ofNullable(currentSnapshot)
                 .map(snapshot -> snapshot.summary().getOrDefault(DEPENDS_ON_TABLES, ""))
                 .orElse("");
+        String dependsOnTableFunctions = Optional.ofNullable(icebergTable.currentSnapshot())
+                .map(snapshot -> snapshot.summary().getOrDefault(DEPENDS_ON_FUNCTIONS, ""))
+                .orElse("");
+
         if (dependsOnTables.isEmpty()) {
+            // If the source of the MV is a PTF then source tables will be empty but source functions won't
+            if (!dependsOnTableFunctions.isEmpty()) {
```

Comment:
move `if (!dependsOnTableFunctions.isEmpty()) {` before `if (dependsOnTables.isEmpty()) {`
it does not have to be inside.